packages/manager/cypress/component/components/ImageSelect/image-select.spec.tsx

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+import { imageFactory } from '@src/factories';
+import React, { useState } from 'react';
+import { mockGetAllImages } from 'support/intercepts/images';
+import { componentTests } from 'support/util/components';
+import { apiMatcher } from 'support/util/intercepts';
+import { randomLabel, randomNumber } from 'support/util/random';
+
+import { ImageSelect } from 'src/components/ImageSelect/ImageSelect';
+
+import type { Image } from '@linode/api-v4';
+
+/**
+ * @param eol - The end-of-life date for the image.
+ * @param deprecated - Whether the image is deprecated.
+ * @returns - A mock image object.
+ */
+const createMockImage = (eol: null | string, deprecated: boolean) => {
+  return imageFactory.build({
+    deprecated,
+    eol,
+    id: `public/${randomNumber()}`,
+    is_public: true,
+    label: randomLabel(),
+    status: 'available',
+  });
+};
+
+/**
+ * @param mockImage - The mock image to intercept and mock.
+ */
+const interceptAndMockImageRequest = (mockImage: { id: string }) => {
+  cy.intercept('GET', apiMatcher('/images/*'), (req) => {
+    req.reply({
+      body: mockImage,
+      headers: { image: mockImage.id },
+    });
+  }).as('mockLinodeRequest');
+};
+
+/**
+ * @param isPast - Whether the date should be in the past.
+ * @param days - The number of days to add to the current date.
+ * @returns - An object containing the date in two formats.
+ */
+const generateDate = (isPast: boolean, days: number) => {
+  const date = new Date();
+  date.setDate(date.getDate() + (isPast ? -days : days));
+
+  const yyyyMMdd = date.toISOString().split('T')[0];
+  const mmddyyyy = `${(date.getMonth() + 1)
+    .toString()
+    .padStart(2, '0')}/${date
+    .getDate()
+    .toString()
+    .padStart(2, '0')}/${date.getFullYear()}`;
+
+  return {
+    mmddyyyy,
+    yyyyMMdd,
+  };
+};
```

Comment:
We can probably handle this a bit more easily using Luxon:

```suggestion
/**
 * @param isPast - Whether the date should be in the past.
 * @param days - The number of days to add to the current date.
 *
 * @returns - An ISO 8601 string containing the date relative to now.
 */
const generateDate = (isPast: boolean, days: number) => {
  // Note: requires importing DateTime from luxon package
  return isPast
    ? DateTime.now().minus({ days }).toISO()
    : DateTime.now().plus({ days }).toISO();
};
```

You can pass this string anywhere you're currently calling `eol.yyyMMdd`, and you can use our `formatDate` util to transform that string anywhere you're currently calling `eol.mmddyy`:

```typescript
// Before:
// getPastDeprecatedWarningMessage(
//   mockPastDeprecatedImage.label,
//   pastDate.mmddyy,
// )
//
// After:
getPastDeprecatedWarningMessage(
  mockPastDeprecatedImage.label,
  formatDate(pastDate, { format: 'MM/dd/yyyy' }),
)
```

This resolves the test failure I'm seeing locally that [Banks had previously mentioned](https://github.com/linode/manager/pull/11570#pullrequestreview-2596365970)

Reply:
Thank you @jdamore-linode, for sharing code. I have added same in this commit https://github.com/linode/manager/pull/11570/commits/cc879f1f9f15c292361d73a1cb4c62e3b1bb42f8, will wait for @bnussman-akamai for confirmation.